pkg/machines/components/diskAdd.jsx

```diff
@@ -477,10 +477,16 @@ class AddDiskModalBody extends React.Component {
         }
 
         // use existing volume
+        let format;
```

Comment:
THanks a lot for finding this, but I think that doesn't belong here. It's a bit confusing the we ignore/overwrite the state's  value for format before submitting the form.

What I think is correct, is to move this piece of code, in the onValueChanged call. And it's time the volume changes, we will get the correct format again. What do you think?

Reply:
Hmm I don't know if we should represent this in state, since this value is at that point not related to state of react component. It's just a value which we pass to libvirt API and can calculate it just before call from other state value.

Reply:
Relocated the logic to onValueChanged, also added commit which resets "Add Disk" dialog everytime user changes modes.